azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -0,0 +1,494 @@
+
+        secret_name = secret_name if secret_name else file_name.replace(".", "-")
+
+        # iterate over secrets to check if secret with same name exists
+        secret_name = self._check_and_update_secret_name(secrets, secret_name, spc_keyvault_name)
+        self._upload_to_key_vault(secret_name, file, cert_extension)
+
+        # check if there is a spc called "opc-ua-connector", if not create one
+        try:
+            opcua_spc = self.ssc_mgmt_client.azure_key_vault_secret_provider_classes.get(
+                resource_group_name=resource_group,
+                azure_key_vault_secret_provider_class_name=OPCUA_SPC_NAME,
+            )
+        except ResourceNotFoundError:
+            opcua_spc = {}
+
+        self._add_secrets_to_spc(
+            secrets=[secret_name],
+            spc=opcua_spc,
+            resource_group=resource_group,
+            spc_keyvault_name=spc_keyvault_name,
+            spc_tenant_id=spc_tenant_id,
+            spc_client_id=spc_client_id,
+        )
+
+        # check if there is a secret sync called "aio-opc-ua-broker-trust-list ", if not create one
+        try:
+            opcua_secret_sync = self.ssc_mgmt_client.secret_syncs.get(
+                resource_group_name=resource_group,
+                secret_sync_name=OPCUA_TRUST_LIST_SECRET_SYNC_NAME,
+            )
+        except ResourceNotFoundError:
+            opcua_secret_sync = {}
+
+        return self._add_secrets_to_secret_sync(
+            secrets=[(secret_name, file_name)],
+            secret_sync=opcua_secret_sync,
+            resource_group=resource_group,
+            spc_name=OPCUA_SPC_NAME,
+            secret_sync_name=OPCUA_TRUST_LIST_SECRET_SYNC_NAME,
+        )
+
+    def issuer_add(
+        self, instance_name: str, resource_group: str, file: str, secret_name: Optional[str] = None
+    ) -> dict:
+        cl_resources = self._get_cl_resources(instance_name=instance_name, resource_group=resource_group)
+        secretsync_spc = self._find_existing_spc(instance_name=instance_name, cl_resources=cl_resources)
+        if not secretsync_spc:
+            return
+
+        # get properties from default spc
+        spc_properties = secretsync_spc.get("properties", {})
+        spc_keyvault_name = spc_properties.get("keyvaultName", "")
+        spc_tenant_id = spc_properties.get("tenantId", "")
+        spc_client_id = spc_properties.get("clientId", "")
+
+        self.keyvault_client = get_keyvault_client(
+            subscription_id=self.default_subscription_id,
+            keyvault_name=spc_keyvault_name,
+        )
+
+        secrets: PageIterator = self.keyvault_client.list_properties_of_secrets()
+
+        # get file extension
+        file_name = os.path.basename(file)
+        cert_extension = validate_file_extension(file_name, [".der", ".crt", ".crl"])
+        # get cert name by removing extension
+        cert_name = file_name.replace(f"{cert_extension}", "")
```

Comment:
note this will replace all instances of the certificate. example: company.crtl.100.crt